### ISSUE TITLE
[css-pseudo-4] Support for highlight pseudos declarations inside @container media queries

### ISSUE THREAD

**Issue by schenney-chromium** [w3c/csswg-drafts#9280]
The interactions between CSS container queries and highlight pseudos are not discussed anywhere, as best I can tell. In chromium right now you cannot define and use a custom highlight pseudo defined inside an @container media query (I haven't checked ::selection or the others). I would like to disallow it due to implementation challenges.

If there's a compelling use case for changing custom highlights depending on containers queries please add it in the comments.

It's also not currently possible to use container based units in highlight declarations. The obvious thing would be to treat them like font units and resolve against the originating element and that's what seems to be said in the resolution to https://github.com/w3c/csswg-drafts/issues/7591. But that's also added implementation complexity.

On the one hand, the most likely thing to drive using container units is the font size which would be available, maybe, once https://github.com/w3c/csswg-drafts/issues/7591 is implemented. On the other hand, do we want the added complexity?

Thoughts?

**Comment by emilio** [comment]
Curious, what are the implementation challenges?

**Comment by emilio** [comment]
Generally these pseudo-elements can't affect layout, so they can be computed once layout is done and all containers are sized...

**Comment by mirisuzanne** [comment]
Is this specific to custom `::highlight()`, or does it also apply to `::selection`? I don't know about the former, but it seems like there would be clear use-cases for changing selection styles in a container.

**Comment by schenney-chromium** [comment]
Custom highlights are the concern. I'll be opening another issue real soon now for other media queries and also looking at the other highlights. For selection, do have any use case in mind for how selection would change based on a container query?

**Comment by mirisuzanne** [comment]
With style queries it's even more likely - but even with a size query I would expect text and background colors to change in many different situations. Any time text or background colors are changing, it would be very likely for `::selection` to change as well - since selection styles are intended to stand out against a given background.

**Comment by lilles** [comment]
I don't understand what the implementation concern for Chrome is here. I think the current issues are just bugs.

I've reported https://crbug.com/1481068 and outlined what needs to be done to fix it and also landed two smaller changes that fixes a lot of the cases.

**Comment by mirisuzanne** [comment]
Given that – I'm adding agenda+ to revisit our resolution (and hopefully revert it). I believe there are clear use-cases for changing these styles in container queries, and there shouldn't be loop concerns.

**Comment by lilles** [comment]
I've fixed the chromium issues, fwiw.

**Comment by schenney-chromium** [comment]
There are still a few issues with the chromium implementation. I have tests that I can add to WPT later this week. But I agree we should revisit and in fact allow highlights inside media queries.

I also think this can be resolved async.

**Comment by astearns** [comment]
The CSSWG will automatically accept this resolution one week from now if no objections are raised here. Anyone can add an emoji to this comment to express support. If you do not support this resolution, please add a new comment.

**Proposed Resolution**: Revert the previous resolution and allow highlight pseudos inside container queries

**Comment by lilles** [comment]
Close as the resolution is to keep the current spec.
